Doc/library/io.rst

```diff
@@ -1147,6 +1147,49 @@ Text I/O
    It inherits from :class:`codecs.IncrementalDecoder`.
 
 
+Static Typing
+-------------
+
+The following protocols can be used for annotating function and method
+arguments for simple stream reading or writing operations. They are decorated
+with :deco:`typing.runtime_checkable`.
+
+.. class:: Reader[T]
+
+   Protocol for reading from a file or other input stream.
+
+   .. versionadded:: next
+
+   .. method:: read(size=..., /)
```

Comment:
This might be slightly cleaner, as there are sort-of two signatures: read all, and read-sized
```suggestion
   .. method:: read()
               read(size, /)
```

Reply:
I would generally agree. But in this case – as it's a protocol – it must be implemented as one function to fulfill the protocol. So there's basically two views on this: The user's who sees this annotation used and wants to supply a matching object, and the implementor's, who actually needs to implement an object with a matching signature. I think in both cases, a single signature is easier to read – especially since other classes in this document also use a single signature for `read()`.

Reply:
In this case I'm just not a fan of `size=..., /`, as it doesn't reflect any runtime semantics. I think the documentation also has the licence to be more precise than is allowed by the runtime -- here, we're using the `.pyi` convention of `...` to signal some unknown value, but the better way to do this is the two distinct signatures. In this vein, there's been talk of multisignatures (e.g. https://discuss.python.org/t/signatures-a-call-to-action/23580, #117671) but nothing done yet. 

However, your broader point that we should be consistent on the signature of `read()` is right, and is out of scope in this PR, so I'm content with leaving it as is for now if you don't want to change it.